File: src/shared/api/system.js
import { invoke } from '@tauri-apps/api/core'

// 获取应用版本信息
export async function getAppVersion() {
  return await invoke('get_app_version')
}


// 检查是否为便携模式
export async function isPortableMode() {
  return await invoke('is_portable_mode')
}

// 启动内置截图功能
export async function startScreenshot() {
  return await invoke('start_builtin_screenshot')
}

// 检查 AI 翻译配置
export async function checkAiTranslationConfig() {
  return await invoke('check_ai_translation_config')
}

// 启用 AI 翻译取消快捷键
export async function enableAiTranslationCancelShortcut() {
  return await invoke('enable_ai_translation_cancel_shortcut')
}

// 禁用 AI 翻译取消快捷键
export async function disableAiTranslationCancelShortcut() {
  return await invoke('disable_ai_translation_cancel_shortcut')
}

// 复制文本
export async function copyTextToClipboard(text) {
  return await invoke('copy_text_to_clipboard', { text })
}

File: src-tauri/src/commands/system.rs
use serde_json::Value;
use clipboard_rs::{Clipboard, ClipboardContext};

// 启动内置截图功能
#[tauri::command]
pub fn start_builtin_screenshot() -> Result<(), String> {
    Ok(())
}

// 检查 AI 翻译配置
#[tauri::command]
pub fn check_ai_translation_config() -> Result<Value, String> {
    use crate::services::get_settings;
    
    let settings = get_settings();
    let is_configured = !settings.ai_api_key.is_empty() 
        && settings.ai_translation_enabled;
    
    Ok(serde_json::json!({
        "is_configured": is_configured,
        "enabled": settings.ai_translation_enabled,
        "api_key_set": !settings.ai_api_key.is_empty(),
    }))
}

// 启用 AI 翻译取消快捷键
#[tauri::command]
pub fn enable_ai_translation_cancel_shortcut() -> Result<(), String> {
    Ok(())
}

// 禁用 AI 翻译取消快捷键
#[tauri::command]
pub fn disable_ai_translation_cancel_shortcut() -> Result<(), String> {
    Ok(())
}

// 复制纯文本
#[tauri::command]
pub fn copy_text_to_clipboard(text: String) -> Result<(), String> {
    let ctx = ClipboardContext::new()
        .map_err(|e| format!("创建剪贴板上下文失败: {}", e))?;
    ctx.set_text(text)
        .map_err(|e| format!("设置剪贴板文本失败: {}", e))
}

